Unable to change Threat Intelligence configuration

When trying to change the configuration on the System -> Configuations page, the following exception is thrown:
```text
2017-09-26 15:39:05,896 ERROR: org.graylog2.rest.resources.system.ClusterConfigResource - Couldn't parse cluster configuration "org.graylog.plugins.threatintel.ThreatIntelPluginConfiguration".
com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of org.graylog.plugins.threatintel.ThreatIntelPluginConfiguration, problem: Null otxApiKey
 at [Source: org.glassfish.jersey.message.internal.EntityInputStream@3018e231; line: 1, column: 95]
	[...]
Caused by: java.lang.NullPointerException: Null otxApiKey
	at org.graylog.plugins.threatintel.AutoValue_ThreatIntelPluginConfiguration$Builder.otxApiKey(AutoValue_ThreatIntelPluginConfiguration.java:119) ~[classes/:?]
	at org.graylog.plugins.threatintel.ThreatIntelPluginConfiguration.create(ThreatIntelPluginConfiguration.java:41) ~[classes/:?]
	[...]
```

--- a/src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java
+++ b/src/main/java/org/graylog/plugins/threatintel/ThreatIntelPluginConfiguration.java
@@ -19,7 +19,8 @@ public abstract class ThreatIntelPluginConfiguration {
     public abstract boolean otxEnabled();
 
     @JsonProperty("otx_api_key")
-    public abstract Optional<String> otxApiKey();
+    @Nullable
+    public abstract String otxApiKey();
 
     @JsonProperty("tor_enabled")
     public abstract boolean torEnabled();
@@ -62,7 +63,7 @@ public abstract class ThreatIntelPluginConfiguration {
     public static abstract class Builder {
         public abstract Builder otxEnabled(boolean otxEnabled);
 
-        public abstract Builder otxApiKey(String otxApiKey);
+        abstract Builder otxApiKey(String otxApiKey);
 
         public abstract Builder torEnabled(boolean torEnabled);
 

--- a/src/main/java/org/graylog/plugins/threatintel/migrations/V20170821100300_MigrateOTXAPIToken.java
+++ b/src/main/java/org/graylog/plugins/threatintel/migrations/V20170821100300_MigrateOTXAPIToken.java
@@ -53,12 +53,12 @@ public class V20170821100300_MigrateOTXAPIToken extends Migration {
         }
 
         final ThreatIntelPluginConfiguration pluginConfig = clusterConfigService.get(ThreatIntelPluginConfiguration.class);
-        if (pluginConfig == null || pluginConfig.otxApiKey() == null || !pluginConfig.otxApiKey().isPresent()) {
+        if (pluginConfig == null || pluginConfig.otxApiKey() == null) {
             clusterConfigService.write(MigrationCompleted.notConvertedKey());
             return;
         }
 
-        final String otxApiKey = pluginConfig.otxApiKey().get();
+        final String otxApiKey = pluginConfig.otxApiKey();
         final DataAdapterDto dataAdapterDto = this.dbDataAdapterService.get(OTX_DATA_ADAPTER_NAME)
                 .orElseThrow(() -> new IllegalStateException("OTX data adapter not present when trying to add API token."));
 
@@ -67,7 +67,8 @@ public class V20170821100300_MigrateOTXAPIToken extends Migration {
         }
 
         final HTTPJSONPathDataAdapter.Config config = (HTTPJSONPathDataAdapter.Config)dataAdapterDto.config();
-        final Map<String, String> newHeaders = new HashMap<>(config.headers() != null ? config.headers() : Collections.emptyMap());
+        final Map<String, String> headers = config.headers();
+        final Map<String, String> newHeaders = new HashMap<>(headers != null ? headers : Collections.emptyMap());
         newHeaders.put("X-OTX-API-KEY", otxApiKey);
         final HTTPJSONPathDataAdapter.Config.Builder updatedConfigBuilder = HTTPJSONPathDataAdapter.Config.builder()
                 .type(config.type())

